processor/src/main/java/org/mapstruct/ap/internal/processor/creation/MappingResolverImpl.java

```diff
@@ -376,32 +377,56 @@ private Assignment resolveViaMethod(Type sourceType, Type targetType, boolean co
             return null;
         }
 
+        /**
+         * Applies matching to given method only
+         *
+         * @param sourceType the source type
+         * @param targetType the target type
+         * @param method     the method to match
+         * @return an assignment if a match, given the criteria could be found. When the method is a
+         * buildIn method, all the bookkeeping is applied.
+         */
+        private Assignment applyMatching(Type sourceType, Type targetType, Method method) {
+
+            if ( method instanceof SourceMethod ) {
```

Comment:
What if the method is not one of these types? It can be an intermediate `ForgedMethod` or an existing `HelperMethod` that we have created. Why shouldn't we reuse that one?

Reply:
is not possible here.. We don't resolve forged methods.. If so, we should throw an unsupported.

Reply:
You are right

Reply:
resolved, I guess?